stacks/cf.py

```diff
@@ -48,9 +49,12 @@ def gen_template(tpl_file, config):
         sys.exit(1)
 
     if len(docs) == 2:
-        return json.dumps(docs[1], indent=2, sort_keys=True), docs[0]
+        tpl, metadata = docs[1], docs[0]
     else:
-        return json.dumps(docs[0], indent=2, sort_keys=True), None
+        tpl, metadata = docs[0], None
+
+    errors = validate_template(tpl)
+    return json.dumps(tpl, indent=2, sort_keys=True), metadata, errors
```

Comment:
Hehe, applying go-style error handling :-)

Reply:
My subconscious is looking toward learning Golang :-)